src/platform_linux.cc

```diff
@@ -154,13 +154,69 @@ std::string GetWorkingDirectory() {
   return working_dir;
 }
 
+// Returns the canonicalized absolute pathname, without expanding symbolic links.
+static optional<std::string> RealPathNotExpandSymlink(std::string path) {
```

Comment:
Can you move this to anonymous namespace section?

Reply:
Done